pkg/server/backuprestoreserver.go

```diff
@@ -391,6 +368,7 @@ func (b *BackupRestoreServer) runEtcdProbeLoopWithSnapshotter(ctx context.Contex
 						b.logger.Warnf("Failed to take first delta snapshot: snapshotter failed with error: %v", err)
```

Comment:
If you take a delta snapshot and it fails because the revision has been compacted then it will return an error. If you get an error the you continue. This means that you are repeatedly attempting to take a delta snapshot which will never succeed, right?

Reply:
yes, you are correct this was the bug. With this PR if delta fails we will continue with full snapshot.